### PR TITLE
[dev 2.7] OPA Gatekeeper 3.10.0

### PR DESCRIPTION
**Pull request by vardhaman22**
This PR adds OPA Gatekeeper 3.10.0 version.

Related issue: https://github.com/rancher/rancher/issues/39438
Adds:

rancher-gatekeeper 101.1.0+up3.10.0
rancher-gatekeeper-crd 101.1.0+up3.10.0

### CODE REVIEW

**Review by rayandas — CHANGES_REQUESTED** [review]
You have to update the rancher mirrored curl image as well in values.yaml. It's not visible in `values.yaml.patch`

**Comment by vardhaman22** [comment]
> You have to update the rancher mirrored curl image as well in values.yaml. It's not visible in `values.yaml.patch`

done please check.

File: index.yaml [hunk 1]
```diff
@@ -5077,6 +5077,36 @@ entries:
     - assets/rancher-external-ip-webhook/rancher-external-ip-webhook-0.1.400.tgz
     version: 0.1.400
   rancher-gatekeeper:
+  - annotations:
+      catalog.cattle.io/auto-install: rancher-gatekeeper-crd=match
+      catalog.cattle.io/certified: rancher
+      catalog.cattle.io/display-name: OPA Gatekeeper
+      catalog.cattle.io/kube-version: '>= 1.20.0-0 < 1.25.0-0'
```
Comment (pennyscissors) [inline index.yaml]: Does this version of gatekeeper support k8s 1.25? I believe we intend to support k8s 1.25 in Rancher 2.7.1, so if this chart supports it we should raise this.

Comment (vardhaman22) [reply index.yaml]: done

**Comment by eliyamlevy** [comment]
@rayandas Can you please check if the changes you requested are fixed? It is blocking the review.

**Comment by rayandas** [comment]
@eliyamlevy yeah it's fixed.

**Review by rayandas — DISMISSED** [review]
LGTM, but wait for @PennyScissors's review before merging.

**Review by pennyscissors — DISMISSED** [review]

**Review by rayandas — APPROVED** [review]

**Pull request merged**